spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastClientConfigResourceCondition.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.hazelcast;
+
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.ResourceCondition;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * {@link SpringBootCondition} used to check if the Hazelcast Client configuration is
+ * available. This either kicks in if a default configuration has been found or if
+ * configurable property referring to the resource to use has been set.
+ *
+ * @author Vedran Pavic
+ * @since 2.0.0
+ */
+public abstract class HazelcastClientConfigResourceCondition extends ResourceCondition {
+
+	static final String CONFIG_SYSTEM_PROPERTY = "hazelcast.client.config";
+
+	protected HazelcastClientConfigResourceCondition(String prefix, String propertyName) {
```

Comment:
Perhaps there is a way to abstract more since all that should change is the name of the file (`hazelcast.xml` vs. `hazelcast-client.xml`).